### ISSUE TITLE
[BUG] SkiaSharp.NativeAssets.Linux.NoDependencies is adding a reference to osx binary which does not exist

### ISSUE THREAD

**Issue by xPaw** [mono/SkiaSharp#1159]
**Description**

We get the following warning in visual studio for our winforms project:
> Warning		The referenced component 'C:\Users\xPaw\.nuget\packages\skiasharp.nativeassets.linux.nodependencies\1.68.1.1\buildTransitive\net45\\..\\..\\runtimes\osx\native\libSkiaSharp.dylib' could not be found.	GUI			

**Basic Information**

- Version with issue:   1.68.1.1
- IDE:   VS 2019
- Platform Target Frameworks:  .net 4.7.2
- Target Devices:  Windows

**Screenshots**

**Reproduction Link**

Project: https://github.com/SteamDatabase/ValveResourceFormat

ValveResourceFormat.csproj (net core) references skia like this:
```
    <PackageReference Include="SkiaSharp" Version="1.68.1.1" />
    <PackageReference Include="SkiaSharp.NativeAssets.Linux.NoDependencies" Version="1.68.1.1" />
```

and the GUI (winforms) references the csproj above.

**Comment by mattleibow** [comment]
Whoah. Let me take a look. That is super weird.

**Comment by bazilshep** [comment]
I have a similar issue, and I think I've tracked down the offending code from the build I'm currently using.

https://github.com/mono/SkiaSharp/blob/7fe55eb3ee6e2dfad48a03851314de8420109dce/binding/SkiaSharp.Desktop/nuget/build/net45/SkiaSharp.targets#L48-L54

This file gets built into the SkiaSharp nuget package and imported into my project's csproj file.
`<Import Project="..\packages\SkiaSharp.1.68.1.1\build\net45\SkiaSharp.targets" Condition="Exists('..\packages\SkiaSharp.1.68.1.1\build\net45\SkiaSharp.targets')" />`

Disabling the `ItemGroup` quoted above in the SkiaSharp.targets file appears to fix the issue for me (disabled via `<ItemGroup Condition="False">` ).

I tried to debug that condition (since it shouldn't be met for my project) by adding some tracing to the package `SkaiSharp.targets` file.
```
<Target Name="TestMessage" AfterTargets="Build" >
  <Message Text="XamarinMacFrameworkRoot" Importance="high"/>
  <Message Text="'$(XamarinMacFrameworkRoot)' != ''" Importance="high"/>
  <Message Text="Full Itemgroup condition" Importance="high"/>
  <Message Text="'$(ShouldIncludeNativeSkiaSharp)' != 'False' and '$(XamarinMacFrameworkRoot)' != '' and '$(TargetFrameworkIdentifier)' != 'Xamarin.Mac' and '$(UseXamMacFullFramework)' == 'true' and ('$(OutputType)' == 'Exe' or '$(IsAppExtension)' == 'true') " Importance="high"/>
</Target>
```
This outputs the following to the build output:
```
2>  XamarinMacFrameworkRoot
2>  '' != ''
2>  Full Itemgroup condition
2>  'True' != 'False' and '' != '' and '.NETFramework' != 'Xamarin.Mac' and '' == 'true' and ('Library' == 'Exe' or '' == 'true') 
```
I'm not sure why its acting like the condition is true, but it is.

**Comment by mattleibow** [comment]
This sounds like a bug in VS. I also started observing this in my IDE and it is just for the `<NativeReference />` elements. I believe Windows was using that for something else. However, this is still a bug in the IDE because it is not supposed to even see that item due to the conditions.

I created a bug report for the IDE: https://developercommunity.visualstudio.com/content/problem/1142283/elements-still-appear-in-the-ide-even-though-they.html

**Comment by Ziriax** [comment]
We get tons of these warnings when building on Windows with SkiaSharp 2.80.2.

I get `access denied` when trying to look at  https://developercommunity.visualstudio.com/content/problem/1142283/elements-still-appear-in-the-ide-even-though-they.html, so I can't check the status.

Is a workaround known? It's driving my colleagues nuts 😉

**Comment by mattleibow** [comment]
Luckily I am a MS employee so I made this public. 😄 Hopefully I don't get into trouble... It might have been me that made it private accidentally.
Anyway, it has not had any progress 😢  due to lack of interest... Probably due to it being private.

Hopefully we can all jump on there and make it go viral.

**Comment by mattleibow** [comment]
Here is a duplicate issue that you can also vote on: https://developercommunity.visualstudio.com/content/problem/1251187/why-nativereference-cannot-recoginze-msbuild-prope.html

**Comment by ju-sti** [comment]
So according to the link above it is fixed in VS 16.11 but I am still seeing the same error in VS2019 (16.11.8) and VS2022. I traced the problem to the Skiasharp.targets file like @bazilshep above and it seems like that the condition is still wonky.

-     Version with issue: 1.68.3
-     IDE: VS 2019+2022
-     Platform Target Frameworks: .net 4.7.2
-     Target Devices: Windows

Thanks for your help

**Comment by mattleibow** [comment]
I believe this was fixed in the IDE because I am no longer seeing this. If not, please reopen the issue on the VS feedback portal because the IDE needs to correctly handle the NativeReference - since that is an official build action.
